src/statsmodels/formula.jl

```diff
@@ -246,78 +235,145 @@ end
 ModelFrame(f::Formula, d::AbstractDataFrame) = ModelFrame(Terms(f), d)
 ModelFrame(ex::Expr, d::AbstractDataFrame) = ModelFrame(Formula(ex), d)
 
+asMatrix(a::AbstractMatrix) = a
+asMatrix(v::AbstractVector) = reshape(v, (length(v), 1))
+
+"""
+    StatsBase.model_response(mf::ModelFrame)
+Extract the response column, if present.  `DataVector` or
+`PooledDataVector` columns are converted to `Array`s
+"""
 function StatsBase.model_response(mf::ModelFrame)
-    mf.terms.response || error("Model formula one-sided")
-    convert(Array, mf.df[round(Bool, mf.terms.factors[:, 1])][:, 1])
+    if mf.terms.response
+        convert(Array, mf.df[mf.terms.eterms[1]])
+    else
+        error("Model formula one-sided")
+    end
 end
 
+"""
+    contr_treatment(n::Integer, contrasts::Bool, sparse::Bool, base::Integer)
+Create a sparse or dense identity of size `n`.  Return the identity if `contrasts`
+is false.  Otherwise drop the `base` column.
+"""
 function contr_treatment(n::Integer, contrasts::Bool, sparse::Bool, base::Integer)
     if n < 2 error("not enought degrees of freedom to define contrasts") end
```

Comment:
It'd be nice to use explicitly typed exceptions, e.g.
```julia
n < 2 && throw(ArgumentError("Not enough degrees of freedom to define contrasts"))
```
Also you have a trailing "t" on "enough."